lib/util.js

```diff
@@ -761,3 +765,94 @@ exports._exceptionWithHostPort = function(err,
   }
   return ex;
 };
+
+function deepEqual(a, b) {
+  return _deepEqual(a, b, false);
+}
+exports.deepEqual = deepEqual;
+
+function deepStrictEqual(a, b) {
+  return _deepEqual(a, b, true);
+}
+exports.deepStrictEqual = deepStrictEqual;
+
+function _deepEqual(a, b, strict) {
+  // All identical values are equivalent, as determined by ===.
+  if (a === b) {
+    return true;
+  } else if (a instanceof Buffer && b instanceof Buffer) {
+    return process.binding('buffer').compare(a, b) === 0;
```

Comment:
It's best to cache the output of `process.binding` in a variable, could you change it to how you did #1171 ?